pkg/secrets/plugins/in-memory/store.go

```diff
@@ -37,36 +23,13 @@ func NewStore() *Store {
 	return s
 }
 
-func (s *Store) Connect(context.Context) error {
-	// Keep track of Connect calls for test asserts later
-	count, err := s.GetConnectCount()
-	if err != nil {
-		return err
-	}
-
-	s.Secrets[MockStoreType][ConnectCount] = strconv.Itoa(count + 1)
-
-	return nil
-}
-
-func (s *Store) Close(ctx context.Context) error {
-	// Keep track of Close calls for test asserts later
-	count, err := s.GetCloseCount()
-	if err != nil {
-		return err
-	}
-
-	s.Secrets[MockStoreType][CloseCount] = strconv.Itoa(count + 1)
-	return nil
-}
-
-func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
+func (s *Store) Resolve(ctx context.Context, keyName string, keyValue string) (string, error) {
 	_, ok := s.Secrets[keyName]
 	if !ok {
 		s.Secrets[keyName] = make(map[string]string, 1)
 	}
 
-	if keyName == secrets.SourceSecret {
+	if keyName == "secret" {
```

Comment:
It creates a circular dependency unfortunately

Reply:
I'm not going to change it now but here's some context about how I made this mess in Porter. :joy: 

I made it easier to create circular dependencies because I am putting test helpers in official packages. The test helpers often reference a particular implementation of an interface, which can lead to a circular dep between that pkg with the implementation and the pkg with the helper. 

If I had created a subpackage for test helpers e.g. pkg/secrets/tests that would have helped avoid this I think (I haven't tried that out though).

I originally didn't make a separate package because in enough cases, the test helper was accessing unexported fields on the original type. If at some point we wanted to review how we write tests, and try out some improvements, I'd totally be up for that.

Reply:
Thanks for the clarification! Yes, I have noticed our tests are in the same package as the actual code. It may help us to catch more things if we have a separate test package too. I would love to slowly get to there as well